### PR TITLE
PM-14805: Use network result in all Retrofit API requests

### PR DESCRIPTION
**Pull request by david-livefront**
## 🎟️ Tracking

[PM-14805](https://bitwarden.atlassian.net/browse/PM-14805)

## 📔 Objective

This PR adds a new class called `NetworkResult` to the app and uses it in place of `Result` for all network requests.

This is done to avoid a `ClassCastException` that can occur where the `Result` in unnecessarily boxed creating a `Result<Result<T>>`. This should not happen with the new `NetworkResult`. 

We will continue to use the regular `Result` everywhere else in the app.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-14805]: https://bitwarden.atlassian.net/browse/PM-14805?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/test/java/com/x8bit/bitwarden/data/platform/datasource/network/retrofit/RetrofitsTest.kt [hunk 1]
```diff
@@ -39,7 +40,7 @@ class RetrofitsTest {
         }
     }
     private val headersInterceptors = mockk<HeadersInterceptor> {
-        mockIntercept { isheadersInterceptorCalled = true }
+        mockIntercept { isHeadersInterceptorCalled = true }
```
Comment (SaintPatrck) [inline app/src/test/java/com/x8bit/bitwarden/data/platform/datasource/network/retrofit/RetrofitsTest.kt:43]: 👍

**Review by SaintPatrck — APPROVED** [review]

**Review by dseverns-livefront — APPROVED** [review]

**Comment by david-livefront** [comment]
Thanks @patrick-bitwarden & @dseverns-livefront

**Comment by david-livefront** [comment]
Thanks @SaintPatrck & @dseverns-livefront

**Pull request merged**